emails/views.py

```diff
@@ -1568,10 +1600,73 @@ def _handle_bounce(message_json: AWS_SNSMessageJSON) -> HttpResponse:
     return HttpResponse("OK", status=200)
 
 
+def _build_disabled_mask_for_spam_email(
+    mask: RelayAddress | DomainAddress, original_spam_email: dict
+) -> EmailMessage:
+    ctx = {
+        "mask": mask.full_address,
+        "spam_email": original_spam_email,
+        "SITE_ORIGIN": settings.SITE_ORIGIN,
+    }
+    html_body = render_to_string("emails/disabled_mask_for_spam.html", ctx)
+    text_body = render_to_string("emails/disabled_mask_for_spam.txt", ctx)
+
+    # Create the message
+    msg = EmailMessage()
+    msg["Subject"] = ftl_bundle.format("relay-disabled-your-mask")
+    msg["From"] = settings.RELAY_FROM_ADDRESS
+    msg["To"] = mask.user.email
+    msg.set_content(text_body)
+    msg.add_alternative(html_body, subtype="html")
+    return msg
+
+
+def _send_disabled_mask_for_spam_email(
+    mask: RelayAddress | DomainAddress, original_spam_email: dict
+) -> None:
+    msg = _build_disabled_mask_for_spam_email(mask, original_spam_email)
+    if not settings.RELAY_FROM_ADDRESS:
+        raise ValueError(
+            "Must set settings.RELAY_FROM_ADDRESS to send disabled_mask_for_spam email."
+        )
+    try:
+        ses_send_raw_email(
+            source_address=settings.RELAY_FROM_ADDRESS,
+            destination_address=mask.user.email,
+            message=msg,
+        )
+    except ClientError as e:
+        logger.error("reply_not_allowed_ses_client_error", extra=e.response["Error"])
+    incr_if_enabled("free_user_reply_attempt", 1)
+
+
+def _disable_masks_for_complaint(message_json: dict) -> None:
+    for destination_address in message_json.get("mail", {}).get("destination", []):
+        try:
+            address = _get_address(destination_address, False)
```

Comment:
(question, blocking)  Why would the `create` for `DomainAddress` be set to `False`? Is this handling for getting a complaint on a `DomainAddress` that does not exist? If so, I think a different error should be raised from `ObjectDoesNotExist`.

Reply:
Yes, it's meant to handle if we ever receive a complaint for a DomainAddress that doesn't exist. E.g.:
1. User creates a domain address
2. User receives some email thru the address
3. User deletes the domain address from Relay
4. User sends a spam complaint about a past email sent thru the domain address

In this scenario, we don't want to re-create the old domain address.

It's raising `ObjectDoesNotExist` just like `_handle_received` and `_reply_allowed` and the other code in `_get_domain_address`. What other error do you think it should raise?

Reply:
(suggestion, non-blocking) `DomainAddress` no longer re-creates deleted DomainAddress: https://github.com/mozilla/fx-private-relay/blob/4cde8436e29149a8fc24073e48b2725b6da6810e/emails/validators.py#L75

If we are handling for getting a complaint on a `DomainAddress` or `RelayAddress` that never existed I think throwing a [SuspiciousOperations](https://docs.djangoproject.com/en/5.1/ref/exceptions/#suspiciousoperation) or [ValidationError](https://docs.djangoproject.com/en/5.1/ref/exceptions/#validationerror) so it is not responding with bad request and separately handle and log in `_disable_masks_for_complaint` as:
```
            logger.error(
                "Received a complaint from a destination address that never existed as "
                "a Relay address.",
            )
```